### PR TITLE
Add Bulgarian Online Store Review Dataset

### PR DESCRIPTION
**Pull request by guangyusong**
## Checklist for adding MMTEB dataset

Reason for dataset addition:

 * Bulgarian for (https://github.com/embeddings-benchmark/mteb/issues/361)
 * This one also has a high average character length, which is rare, especially for non-English languages.

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] I have considered the size of the dataset and reduced it if it is too big (2048 examples is typically large enough for most tasks)
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [POINTS.md](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) file.

### CODE REVIEW

File: mteb/tasks/Classification/bul/BulgarianStoreReviewSentimentClassfication.py
Comment (isaac-chung) [inline mteb/tasks/Classification/bul/BulgarianStoreReviewSentimentClassfication.py:19]: fix indent.

File: mteb/tasks/Classification/bul/BulgarianStoreReviewSentimentClassfication.py
Comment (isaac-chung) [inline mteb/tasks/Classification/bul/BulgarianStoreReviewSentimentClassfication.py:46]: label here should be of type `int` (see `class AbsTaskClassification`), where as this HF dataset's `Category` column is of type string. Each category needs to be mapped to an `int`.

**Review by isaac-chung — CHANGES_REQUESTED** [review]
Thanks for your contribution! Added a few comments below.

File: mteb/tasks/Classification/bul/BulgarianStoreReviewSentimentClassfication.py
Comment (guangyusong) [reply mteb/tasks/Classification/bul/BulgarianStoreReviewSentimentClassfication.py]: Thanks, addressed!

File: mteb/tasks/Classification/bul/BulgarianStoreReviewSentimentClassfication.py
Comment (imenelydiaker) [reply mteb/tasks/Classification/bul/BulgarianStoreReviewSentimentClassfication.py]: For class encoding, you can use `class_encode_columns()` from `datasets` (see docs [here](https://huggingface.co/docs/datasets/v2.18.0/en/package_reference/main_classes#datasets.Dataset.class_encode_column))

Comment (imenelydiaker) [inline mteb/tasks/Classification/bul/BulgarianStoreReviewSentimentClassfication.py]:
```suggestion
        self.dataset = self.dataset.rename_columns({"Review": "text", "Category": "label"})
```

Comment (guangyusong) [reply mteb/tasks/Classification/bul/BulgarianStoreReviewSentimentClassfication.py]: Thanks for the example!

Comment (guangyusong) [reply mteb/tasks/Classification/bul/BulgarianStoreReviewSentimentClassfication.py]: Thanks, it wouldn't let me press the commit suggestion button but I pushed in a new commit.

**Comment by KennethEnevoldsen** [comment]
Note the potentially changed scoring system format in #438.

**Comment by isaac-chung** [comment]
Let's hold off until #438 is merged.

**Comment by KennethEnevoldsen** [comment]
#438 is in!

@isaac-chung if everything looks good here feel free to merge it in (you can do it using a temporary branch similar to what I did for #376).

**Pull request merged**
